src/rules/noInvalidTemplateStringsRule.ts

```diff
@@ -50,10 +50,28 @@ function walk(ctx: Lint.WalkContext<void>) {
     });
 
     function check(node: ts.StringLiteral): void {
-        const idx = node.text.search(/\$\{/);
-        if (idx !== -1) {
+        /**
+         * Finds instances of '${'
+         */
+        const findTemplateString = new RegExp(/\$\{/);
+            
+        const index = node.text.search(findTemplateString);
+        if (index !== -1) {
```

Comment:
this needs to be a loop. otherwise you only check the first occurence and don't look further

this one should result in an error: ``"\${foo}bar${baz}"